applications/StructuralMechanicsApplication/custom_processes/compute_moment_of_inertia_process.h

```diff
@@ -0,0 +1,249 @@
+#define KRATOS_COMPUTE_MOMENT_OF_INERTIA_PROCESS
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "processes/process.h"
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+/**
+ * @class ComputeMomentOfInertiaProcess
+ *
+ * @ingroup StructuralMechanicsApplication
+ *
+ * @brief This method computes the moment of inertia (rotational)
+ * @details It takes into account all elements in the ModelPart
+ *
+ * @author Philipp Bucher, Salman Yousaf
+*/
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) ComputeMomentOfInertiaProcess
+    : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ComputeMomentOfInertiaProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ComputeMomentOfInertiaProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    ComputeMomentOfInertiaProcess(
+        ModelPart& rThisModelPart,
+        const Point& rPoint1,
+        const Point& rPoint2
+        ):mrThisModelPart(rThisModelPart) , mrPoint1(rPoint1), mrPoint2(rPoint2)
+    { }
+
+    /// Destructor.
+    ~ComputeMomentOfInertiaProcess() override = default;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Execute() override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ComputeMomentOfInertiaProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "ComputeMomentOfInertiaProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrThisModelPart; // The main model part
+    const Point& mrPoint1;      // Point 1 of the axis of rotation
```

Comment:
Principal axis?

Reply:
is defined by the two points given as input

Reply:
For defining properly the principal axis you will need three points, otherwise I assume that you are considering 0,0

Reply:
hm then I don't get what you mean by principal axis

a line (an axis) is fully defined with two points

Reply:
Yes, because now I know it is the mass moment of inertia, for the area moment of inertia (second moment of inertia) you will need three

![imagen](https://user-images.githubusercontent.com/19991680/49148127-ca0cf400-f306-11e8-8992-f6530e835de0.png)

Reply:
ah ok, sorry for the confusion